### PR TITLE
fix(findExports): correctly dedup named exports

### PR DESCRIPTION
**Pull request by antfu**
close https://github.com/nuxt/framework/issues/7658

### CODE REVIEW

File: src/analyze.ts [hunk 1]
```diff
@@ -135,15 +135,14 @@ export function findExports (code: string): ESMExport[] {
...
 
-  return exports.filter((exp, index, exports) => {
+  return exports
     // Filter false positive export matches
-    if (exportLocations && !_isExportStatement(exportLocations, exp)) {
-      return false
-    }
+    .filter(exp => !exportLocations || _isExportStatement(exportLocations, exp))
     // Prevent multiple exports of same function, only keep latest iteration of signatures
-    const nextExport = exports[index + 1]
```
Comment (antfu) [inline src/analyze.ts:144]: We were dedupe with the one that filtered out by AST. Changed to two passes of `.filter` to ensure `exports` is the real exports we want for dedupe.

File: src/analyze.ts [hunk 2]
```diff
@@ -179,7 +178,10 @@ interface TokenLocation {
 }
 
 function _isExportStatement (exportsLocation: TokenLocation[], exp: ESMExport) {
-  return exportsLocation.some(location => exp.start <= location.start && exp.end >= location.end)
+  return exportsLocation.some(location =>
+    (exp.start <= location.start && location.start <= exp.end) ||
+    (exp.start <= location.end && location.end <= exp.end)
```
Comment (pi0) [inline src/analyze.ts:183]: Previous condition (token _within_ exp) seems keep tests pass. Why we need to change it to (token start within match || token end within match)?

Comment (pi0) [reply src/analyze.ts]: Made it into a comment to marge fix asap. fbf5c419df1d938286934b745d4be14104d3514d. Would be happy to discuss more and change checking strategy if you think there is a case needs this check.

**Review by pi0 — APPROVED** [review]

File: src/analyze.ts (hunk 2 above)
Comment (antfu) [reply src/analyze.ts]: That was my initial assumption of the cause - yes I think we don't need to change it for this case.

**Pull request merged**